### PR TITLE
[MDS-5352] Close ESUP permits

### PR DESCRIPTION
**Pull request by asinn134**
## Objective 

[MDS-5352](https://bcmines.atlassian.net/browse/MDS-5352)

- Adjusted the existing ExplosivesPermitCloseForm component to be more in line with the mockup
- Changed the date input to use the timezone date input
- Added closed_by column to the explosives_permit table and saving IDIR user name to column
- Updated close esup flow to include amendments
- Fixed return of permit amendments to be ordered by id (this is a sequential field)
- created `explosivesPermitAmendmentActionCreator`

image

### CODE REVIEW

File: services/core-web/src/components/mine/ExplosivesPermit/ExplosivesPermit.tsx [hunk 1]
```diff
@@ -73,15 +75,25 @@ export const ExplosivesPermit: FC<ExplosivesPermitProps> = ({
   };
 
   const handleUpdateExplosivesPermit = (values) => {
+    console.log("values", values);
```
Comment (taraepp) [inline services/core-web/src/components/mine/ExplosivesPermit/ExplosivesPermit.tsx]: oops

**Review by henryoforeh-dev — DISMISSED** [review]

**Review by isuru-aot — DISMISSED** [review]

File: services/core-web/src/components/Forms/ExplosivesPermit/ExplosivesPermitCloseForm.js [hunk 2]
```diff
@@ -18,42 +18,33 @@ const propTypes = {
...
-              label="Date Permit was Closed"
-              component={RenderDate}
+              normalize={normalizeDatetime}
+              component={RenderDateTimeTz}
+              validate={[
+                dateNotInFutureTZ,
+                required,
+                dateTimezoneRequired("esup_permit_close_timezone"),
+              ]}
+              props={{ timezoneFieldProps: { name: "esup_permit_close_timezone" } }}
```
Comment (taraepp) [inline services/core-web/src/components/Forms/ExplosivesPermit/ExplosivesPermitCloseForm.js:47]: In hindsight, all this complexity could have been avoided if we saved if we saved it as a "YYYY-MM-DD" type string instead. But I do like the normalize function, isn't it cool?

Comment (matbusby-fw) [reply services/core-web/src/components/Forms/ExplosivesPermit/ExplosivesPermitCloseForm.js:47]: 😁 So cool

**Review by taraepp — APPROVED** [review]
LGTM

**Review by henryoforeh-dev — APPROVED** [review]
Nice work!

**Pull request merged**